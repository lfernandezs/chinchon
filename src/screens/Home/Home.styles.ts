import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#fff',
		alignItems: 'center',
		justifyContent: 'center',
	},
	title: {
		fontSize: 40,
		fontWeight: 'bold',
		marginBottom: 20,
	},
	button: {
		backgroundColor: '#000',
		padding: 10,
		borderRadius: 5,
		marginTop: 10,
	},
	buttonLabel: {
		color: '#fff',
		fontSize: 20,
	},
});

export default styles;
