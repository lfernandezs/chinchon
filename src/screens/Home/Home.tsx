import React from 'react';
import { Pressable, Text, View } from 'react-native';
import useHomePresenter from '../../presenters/HomePresenter/useHomePresenter';
import styles from './Home.styles';

const Home = () => {
	const presenter = useHomePresenter();
	return (
		<View style={styles.container}>
			<Text testID={'title'} style={styles.title}>
				{presenter.screenTitle}
			</Text>
			<Pressable
				testID={'new-game'}
				style={styles.button}
				onPress={presenter.onButtonPress}
			>
				<Text style={styles.buttonLabel}>{presenter.buttonLabel}</Text>
			</Pressable>
		</View>
	);
};

export default Home;
