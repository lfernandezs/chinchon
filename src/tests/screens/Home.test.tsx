import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';
import Home from '../../screens/Home/Home';

const mockNavigate = jest.fn();

jest.mock('@react-navigation/native', () => ({
	useNavigation: () => ({
		navigate: mockNavigate,
	}),
}));

describe('Home', () => {
	it('should render correctly', () => {
		const screen = render(<Home />);

		expect(screen).toMatchSnapshot();
	});

	describe('when the screen is rendered', () => {
		it('should render the title', () => {
			const { getByTestId } = render(<Home />);

			expect(getByTestId('title')).toBeTruthy();
			expect(getByTestId('title')).toHaveTextContent('ChinchÃ³n');
		});

		it('should render the button to start a new game', () => {
			const { getByTestId } = render(<Home />);

			expect(getByTestId('newGame')).toBeTruthy();
			expect(getByTestId('newGame')).toHaveTextContent('Nueva partida');
		});
	});

	describe('when the button to start a new game is pressed', () => {
		it('should navigate to the NewGame screen', async () => {
			const { getByTestId } = render(<Home />);
			const button = getByTestId('newGame');
			fireEvent.press(button);
			expect(mockNavigate).toHaveBeenCalled();
		});
	});
});
