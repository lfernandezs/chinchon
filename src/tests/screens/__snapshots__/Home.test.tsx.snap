// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 40,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
    testID="title"
  >
    Chinch√≥n
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000",
        "borderRadius": 5,
        "marginTop": 10,
        "padding": 10,
      }
    }
    testID="new-game"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
        }
      }
    >
      Nueva partida
    </Text>
  </View>
</View>
`;
